packages/auth/src/app.ts

```diff
@@ -367,7 +367,9 @@ export class App {
   }
 
   private async createKoaServer(): Promise<Koa<Koa.DefaultState, AppContext>> {
-    const koa = new Koa<DefaultState, AppContext>()
+    const koa = new Koa<DefaultState, AppContext>({
+      proxy: this.config.trustProxy
+    })
```

Comment:
Should we make sure that also only happens in the `autopeer` environment?

Reply:
this is an environment variable with default value to `false`
enabling it will allow koa application to check `X-Forwarded-Proto` header when determining the request protocol.
We are going to use it on rafiki.money to fix the issue with requests that are redirected as `http`.
It should be available in all environments IMO